tensorrt_llm/_torch/attention_backend/fmha/msa_proxy_mqa.py

```
@@ -0,0 +1,167 @@
+        # (~1/3 of processes emit garbage logits mid-decode). The real
+        # import is deferred to first kernel use, by which point
+        # flashinfer has initialized tvm_ffi itself.
+        if importlib.util.find_spec("fmha_sm100") is None:
+            logger.debug("MsaProxyMqaFmha is unavailable: fmha_sm100 package not installed.")
+            return False
+        if not torch.cuda.is_available():
+            logger.debug("MsaProxyMqaFmha is unavailable: no CUDA device.")
+            return False
+        try:
+            major, _ = torch.cuda.get_device_capability()
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)